jsonapi/marshal.go

```diff
@@ -446,9 +457,9 @@ func marshalStruct(data MarshalIdentifier, information ServerInformation) (*Docu
 }
 
 func getStructType(data interface{}) string {
-	entityName, ok := data.(EntityNamer)
-	if ok {
-		return entityName.GetName()
+	identifier, ok := data.(MarshalIdentifier)
```

Comment:
I think that getStructType should accept MarshalIdentifier as argument, so you no need to cast type here

Reply:
I need to perform casting to MarshalIdentifier in getStructType or in the checkType. Should I return to double casting in the checkType?